mapie/tests/test_regression.py

```diff
@@ -857,22 +879,125 @@ def test_fit_parameters_passing() -> None:
+    for estimator in mapie.estimator_.estimators_:
+        assert estimator.estimators_.shape[0] == 3
 
-    mapie.fit(X, y, monitor=early_stopping_monitor)
 
-    assert mapie.estimator_.single_estimator_.estimators_.shape[0] == 3
+def test_predict_parameters_passing() -> None:
+    """
+    Test passing predict parameters.
+    Checks that y_pred from train are 0, y_pred from test are 0 and
+    we check that y_pred constructed with or without predict_params
+    are different
+    """
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    score = AbsoluteConformityScore(sym=True)
+    mapie_1 = MapieRegressor(estimator=custom_gbr, conformity_score=score)
+    mapie_2 = MapieRegressor(estimator=custom_gbr, conformity_score=score)
+    predict_params = {'check_predict_params': True}
+    mapie_1 = mapie_1.fit(
+        X_train, y_train, predict_params=predict_params
+    )
+    mapie_2 = mapie_2.fit(X_train, y_train)
+    y_pred_1 = mapie_1.predict(X_test, **predict_params)
+    y_pred_2 = mapie_2.predict(X_test)
+    np.testing.assert_allclose(mapie_1.conformity_scores_, np.abs(y_train))
+    np.testing.assert_allclose(y_pred_1, 0)
+    with np.testing.assert_raises(AssertionError):
+        np.testing.assert_array_equal(y_pred_1, y_pred_2)
 
-    for estimator in mapie.estimator_.estimators_:
+
+def test_fit_params_expected_behavior_unaffected_by_predict_params() -> None:
+    """
+    We want to verify that there are no interferences
+    with predict_params on the expected behavior of fit_params
+    Checks that underlying GradientBoosting
+    estimators have used 3 iterations only during boosting,
+    instead of default value for n_estimators (=100).
+    """
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    mapie_1 = MapieRegressor(estimator=custom_gbr)
+    mapie_2 = MapieRegressor(estimator=custom_gbr)
+    fit_params = {'monitor': early_stopping_monitor}
+    predict_params = {'check_predict_params': True}
+    mapie_1 = mapie_1.fit(
+        X_train, y_train,
+        fit_params=fit_params, predict_params=predict_params
+    )
+    mapie_2 = mapie_2.fit(X_train, y_train, predict_params=predict_params)
+
+    assert mapie_1.estimator_.single_estimator_.estimators_.shape[0] == 3
+    for estimator in mapie_1.estimator_.estimators_:
         assert estimator.estimators_.shape[0] == 3
+    assert (mapie_2.estimator_.single_estimator_.n_estimators ==
+           custom_gbr.n_estimators)
+    for estimator in mapie_2.estimator_.estimators_:
+        assert estimator.n_estimators == custom_gbr.n_estimators
+
+
+def test_predict_params_expected_behavior_unaffected_by_fit_params() -> None:
```

Comment:
Same logic as previously mentionned. Here we only need to test that the `fit_params` don't affect the `predict_params` so, no need for a second MAPIE model.